src/test/java/org/broadinstitute/hellbender/engine/dataflow/DataflowTestUtils.java

```diff
@@ -0,0 +1,123 @@
+package org.broadinstitute.hellbender.engine.dataflow;
+
+import com.google.cloud.dataflow.sdk.Pipeline;
+import com.google.cloud.dataflow.sdk.testing.DataflowAssert;
+import com.google.cloud.dataflow.sdk.transforms.Create;
+import com.google.cloud.dataflow.sdk.transforms.SerializableFunction;
+import com.google.cloud.dataflow.sdk.values.KV;
+import com.google.cloud.dataflow.sdk.values.PCollection;
+import com.google.common.collect.Lists;
+import org.apache.commons.collections4.CollectionUtils;
+import org.broadinstitute.hellbender.engine.dataflow.datasources.ReadContextData;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.testng.Assert;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataflowTestUtils {
+    public static <T> PCollection<T> PCollectionCreateAndVerify(Pipeline p, Iterable<T> list) {
+        Iterable<T> copy = Lists.newArrayList(list.iterator());
+        Assert.assertEquals(list, copy);
+        PCollection<T> pCollection = p.apply(Create.of(list));
+        DataflowAssert.that(pCollection).containsInAnyOrder(copy);
+        return pCollection;
+    }
+
+    public static class KeyIterableValueChecker<K, V> implements SerializableFunction<Iterable<KV<K, Iterable<V>>>, Void> {
+        private static final long serialVersionUID = 1l;
+
+        private final List<KV<K, List<V>>> expectedValues;
+
+        public KeyIterableValueChecker( final Iterable<KV<K, Iterable<V>>> expectedValues ) {
+            this.expectedValues = unpackKVIterable(expectedValues);
+        }
+
+        @Override
+        public Void apply( Iterable<KV<K, Iterable<V>>> input ) {
+            final List<KV<K, List<V>>> actualValues = unpackKVIterable(input);
+
```

Comment:
Why not just call CollectionUtils.isEqualCollection(expectedValues, actualValues) here?